fc/parsing/actions.py

```diff
@@ -1200,204 +1294,341 @@ def _number_generator(self, value, units):
-                    raise ProtocolError(
-                        f'Units must be specified for input variables not appearing in the model;'
-                        ' none are given for {var.prefixed_name}.')
-
-                # Add the new input variable, with ontology annotation
-                variable = self._create_variable(var)
+                # Variable not found: Still OK, as long as it refers to an existing model variable
+                try:
+                    self._variable_generator(name)
+                except MissingVariableError:
+                    raise ProtocolError(f'Define or clamp statement found for unknown variable: {name}.')
 
-            # Store initial value if given
-            if var.initial_value is not None:
-                self.initial_values[var.rdf_term] = var.initial_value
+                # Create new protocol variable to store info
+                pvar = name_to_pvar[name] = ProtocolVariable(name)
 
-            # Maintain link to time variable, if needed
-            is_time = variable is self.time_variable
+            # Store equation
+            pvar.update(equation=eq)
 
-            # Convert units if needed
-            if var.units is not None:
-                units = self.units.get_unit(var.units)
-                if units != variable.units:
-                    variable = self.model.convert_variable(variable, units, DataDirectionFlow.INPUT)
+        # Resolve references to model variables
+        # If multiple references point to the same model variable, merge them
+        var_to_pvar = {}
+        aliases = []
+        for pvar in name_to_pvar.values():
+            try:
+                pvar.model_variable = self._variable_generator(pvar.name)
+            except MissingVariableError:
+                # Could be optional, or vector output (which we can only determine post model modifications)
+                continue
 
-                    # Update cached time variable
-                    if is_time:
-                        self.time_variable = variable
+            # Check if another reference already points to this model variable
+            try:
+                partner = var_to_pvar[pvar.model_variable]
+            except KeyError:
+                var_to_pvar[pvar.model_variable] = pvar
+                continue
 
-    def _add_local_variables(self):
-        """
-        Modify the model by adding any variables defined in ``var`` statements.
-        """
-        for var in self.local_var_declarations:
+            # Merge, and mark pvar for removal
+            partner.merge(pvar)
+            aliases.add(pvar)
 
-            # Variable names must be unique (can't even be re-used in imported/nested protocols)
-            if var.name in self.local_vars:
-                raise ProtocolError(f'Variable "{var.name}" was defined by more than one var statement.')
+        # Remove 'alias' references
+        for pvar in aliases:
+            del name_to_pvar[pvar.name]
```

Comment:
```suggestion
            del name_to_pvar[pvar.name]
        del aliases
```
To make clear that these are not needed in the rest of the method, but `var_to_pvar` is.